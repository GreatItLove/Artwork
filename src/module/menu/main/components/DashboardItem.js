import React from 'react';
import { ListItem } from 'material-ui/List';
import { NavLink } from 'react-router-dom';
import { translate } from 'react-i18next';
import PropTypes from 'prop-types';

import { ADMIN_PANEL_BASE_PATH } from '../../../../config/routesPatch';

const propTypes = {
  style: PropTypes.object.isRequired,
  t: PropTypes.func.isRequired
};

/**
 * @function DashboardItem
 * @param {Object} style
 * @param {Function} t translate function
 * @returns {XML}
 */
function DashboardItem({ style, t }) {
  return (
    <ListItem
      key="dashboard"
      primaryText={t('menuItems.dashboard')}
      innerDivStyle={style}
      leftIcon={<i className="material-icons">dashboard</i>}
      containerElement={<NavLink to={ADMIN_PANEL_BASE_PATH} exact />}
    />
  );
}

DashboardItem.propTypes = propTypes;

export { DashboardItem as DashboardItemTest };

export default translate('app')(DashboardItem);
