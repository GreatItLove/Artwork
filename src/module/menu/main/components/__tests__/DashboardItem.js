import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { DashboardItemTest as DashboardItem } from '../DashboardItem';

describe('DashboardItem', () => {
  it('Snapshot of DashboardItem', () => {
    const style = {};
    const t = jest.fn();
    const wrapper = shallow(<DashboardItem {...{ style, t }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
