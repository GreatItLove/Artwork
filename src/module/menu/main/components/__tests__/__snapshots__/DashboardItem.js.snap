// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardItem Snapshot of DashboardItem 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement={
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      exact={true}
      to="/"
    />
  }
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  innerDivStyle={Object {}}
  insetChildren={false}
  leftIcon={
    <i
      className="material-icons"
    >
      dashboard
    </i>
  }
  nestedItems={Array []}
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryTogglesNestedList={false}
  secondaryTextLines={1}
/>
`;
